### PR TITLE
Fix typescript issue

### LINKED ISSUE #194: at-loader errors for d.ts files
Hi,
typescript loader gives me these errors when I use aurelia-form with the typescript-webpack navigation skeleton. The aurelia app works fine.

> webpack: Compiled successfully.
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:216:1
>     TS1128: Declaration or statement expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:216:16
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:218:8
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:219:1
>     TS1128: Declaration or statement expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:219:16
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:221:8
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:222:1
>     TS1128: Declaration or statement expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:222:16
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:224:8
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:225:1
>     TS1128: Declaration or statement expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:225:16
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:227:8
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:228:1
>     TS1128: Declaration or statement expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:228:16
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:230:8
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:231:1
>     TS1128: Declaration or statement expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:231:16
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:233:8
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:234:1
>     TS1128: Declaration or statement expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:234:16
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:236:8
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:237:1
>     TS1128: Declaration or statement expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:237:16
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:239:8
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:240:1
>     TS1128: Declaration or statement expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:240:16
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:242:8
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:243:1
>     TS1128: Declaration or statement expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:243:16
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:245:8
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:246:1
>     TS1128: Declaration or statement expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:246:16
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:248:8
>     TS1005: ';' expected.
> 
> [at-loader] ./node_modules/aurelia-config/dist/aurelia-config.d.ts:24:16
>     TS1030: 'export' modifier already seen.
> 
> [at-loader] ./node_modules/aurelia-config/dist/aurelia-config.d.ts:49:2
>     TS1036: Statements are not allowed in ambient contexts.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:6:9
>     TS2305: Module '"C:/Git/pappt/aurelia-form-demo-typescript-webpack/node_modules/aurelia-config/dist/aurelia-config"' has no exported member 'Configuration'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:10:15
>     TS2307: Cannot find module 'aurelia-form/decorator/index'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:216:8
>     TS2304: Cannot find name 'declare'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:218:3
>     TS2304: Cannot find name 'from'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:219:8
>     TS2304: Cannot find name 'declare'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:221:3
>     TS2304: Cannot find name 'from'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:222:8
>     TS2304: Cannot find name 'declare'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:224:3
>     TS2304: Cannot find name 'from'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:225:8
>     TS2304: Cannot find name 'declare'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:227:3
>     TS2304: Cannot find name 'from'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:228:8
>     TS2304: Cannot find name 'declare'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:230:3
>     TS2304: Cannot find name 'from'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:231:8
>     TS2304: Cannot find name 'declare'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:233:3
>     TS2304: Cannot find name 'from'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:234:8
>     TS2304: Cannot find name 'declare'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:236:3
>     TS2304: Cannot find name 'from'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:237:8
>     TS2304: Cannot find name 'declare'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:239:3
>     TS2304: Cannot find name 'from'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:240:8
>     TS2304: Cannot find name 'declare'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:242:3
>     TS2304: Cannot find name 'from'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:243:8
>     TS2304: Cannot find name 'declare'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:245:3
>     TS2304: Cannot find name 'from'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:246:8
>     TS2304: Cannot find name 'declare'.
> 
> [at-loader] ./node_modules/aurelia-form/dist/aurelia-form.d.ts:248:3
>     TS2304: Cannot find name 'from'.
> 
> [at-loader] ./node_modules/aurelia-view-manager/dist/aurelia-view-manager.d.ts:87:31
>     TS2304: Cannot find name 'ViewEngine'.
> 
> [at-loader] ./node_modules/aurelia-view-manager/dist/aurelia-view-manager.d.ts:87:63
>     TS2304: Cannot find name 'ViewCompileInstruction'.
> 
> [at-loader] ./node_modules/aurelia-view-manager/dist/aurelia-view-manager.d.ts:87:101
>     TS2304: Cannot find name 'ResourceLoadContext'.
> 
> [at-loader] ./node_modules/aurelia-view-manager/dist/aurelia-view-manager.d.ts:87:131
>     TS2304: Cannot find name 'ViewFactory'.

### PR DESCRIPTION
**Pull request by jvkassi**
Hi,

@RWOverdijk, i made the same config as of 
https://github.com/SpoonX/aurelia-form/pull/204
I had to read doc of typescript to get my head around that.

Point me if something wrong.
This is also needed to fix: https://github.com/SpoonX/aurelia-form/issues/194

### CODE REVIEW

**Comment by RWOverdijk** [comment]
Looks good, but one question. Why the concat? I don't think we want that, as it will put everything in a single file, which is frustrating when debugging.

**Comment by jvkassi** [comment]
Yeah, but i mean most of the libraries does that. It's not that much of a big file.
The concat helps for the type definition

**Comment by RWOverdijk** [comment]
Alright then. I'll publish a new version soon. :)

**Review by RWOverdijk — APPROVED** [review]

File: src/plugin-manager.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import {FrameworkConfiguration} from 'aurelia-framework';
  * @export
  * @interface PluginDefinition
  */
-export interface PluginDefinition {
+interface PluginDefinition {
```
Comment (RWOverdijk) [inline src/plugin-manager.js:11]: Why did you remove the export? That instantly makes it a BC break :(

**Review by RWOverdijk — CHANGES_REQUESTED** [review]

File: src/plugin-manager.js (hunk 1 above)
Comment (jvkassi) [reply src/plugin-manager.js:11]: It was causing this repetition 
https://github.com/SpoonX/aurelia-config/blob/master/dist/aurelia-config.d.ts#L24

This should be the fix ? 
https://github.com/Microsoft/TypeScript/issues/3194#issuecomment-103080467

Comment (RWOverdijk) [reply src/plugin-manager.js:11]: It's not the same as a default export, is it? Or does typescript export all interfaces anyway?

Comment (jvkassi) [reply src/plugin-manager.js:11]: It seems to me typescript export all interfaces anyway, cause it exported on the definition.
But i am not too sure about that

Comment (jvkassi) [reply src/plugin-manager.js:11]: @RWOverdijk  can you test it too ?

Comment (RWOverdijk) [reply src/plugin-manager.js:11]: I'm really trying to finish my projects so I can give this a go. Sorry for the delay (on both PRs).

Comment (jvkassi) [reply src/plugin-manager.js:11]: @RWOverdijk  this one also :smile:

**Comment by RWOverdijk** [comment]
I can't even remember this PR. Have you been using this?

**Comment by jvkassi** [comment]
yeah, it's still working in my dev environment

**Review by RWOverdijk — APPROVED** [review]

**Comment by RWOverdijk** [comment]
I'll make a prerelease again. Once both are tested well, it can be a major release :)

**Comment by RWOverdijk** [comment]
Done :)

**Comment by jvkassi** [comment]
Awesome !

**Pull request merged**